### PR TITLE
Patterns for cleanup of Client.rb

### PR DESCRIPTION
**Pull request by eanders**
## _Merging this PR_
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
* This attempts to define some patterns for compartmentalizing some of the code in Client.rb
* Defines some PORO to handle enrollment calculations and to support the enrollment view on the client dashboard
* Moves more CAS related code into the CAS concern

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
* Code clean-up

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/models/client_history/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+class ClientHistory::Calculator
+  attr_accessor :client
+  def initialize(client:)
+    self.client = client
+  end
+
+  # If we haven't been in a literally homeless project type (ES, SH, SO) in the last 30 days, this is a new episode
+  # You aren't currently housed in PH, and you've had at least a week of being housed in the last 90 days
+  def new_episode?(enrollments:, enrollment:)
```
Comment (eanders) [inline app/models/client_history/calculator.rb]: Investigate if this can be combined with `enrollment.new_episode?`

Comment (eanders) [reply app/models/client_history/calculator.rb]: Add a test for `new_episode?`

**Comment by eanders** [comment]
@gigxz, @ttoomey and I spoke about this, but it would be great to get your feedback.  This is a first pass at removing some low hanging fruit from the client model with the intent of slowly working through the model to decrease the overall size and compartmentalize the logic.

File: app/models/client_history/calculator.rb (hunk 1 above)
Comment (gigxz) [reply app/models/client_history/calculator.rb]: `(enrollments:, enrollment:)` is hard to wrap my head around, maybe we could change at least the internal API to have more descriptive parameter names?

File: app/models/client_history/enrollment_view.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class ClientHistory::EnrollmentView
```
Comment (gigxz) [inline app/models/client_history/enrollment_view.rb:7]: Maybe a comment for what should go in this class? Eg is it for helpers specifically for the `clients/enrollment_table` view?

**Comment by gigxz** [comment]
I think this looks great @eanders, I like that direction.

My only thoughts would be:
* Should we take this time to add comments to the PORO classes (/public methods on them) so we know what they are for? (I just have no context on these)
* Do we have enough test coverage?

File: app/models/client_history/calculator.rb (hunk 1 above)
Comment (eanders) [reply app/models/client_history/calculator.rb]: @gigxz how about using `residential_enrollments:, enrollment:`? (as it is now?)

**Comment by eanders** [comment]
@gigxz turns out we do have test coverage, and I'd broken some with the last change.  I'm good merging when you are.

**Review by gigxz — APPROVED** [review]

**Pull request merged**